### PR TITLE
support processing other packet from ipstack

### PR DESCRIPTION
**Pull request by xmh0511**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,6 +34,7 @@ tun2 = { version = "0.6", features = ["async"] }
 tokio = { version = "1.35", features = [
     "rt-multi-thread",
 ], default-features = false }
+packet = "0.1.4"
```
Comment (SajjadPourali) [inline Cargo.toml:37]: I am hesitant to use this dependency since it is left unmaintained.

**Comment by SajjadPourali** [comment]
It is better to create two types for the `IpStackStream`: one for `UnknownTransport`, which we know is `IP` but is neither `TCP` nor `UDP`, and another for `UnknowNetwork` for anything that is not IP. I believe these types allow users to implement their protocols manually.
-----
Since ICMP is not primarily designed to transfer data, we have to create an enum that supports the most useful types of ICMP in the IpStack usage scenarios, such as Echo.

**Comment by SajjadPourali** [comment]
Something like this?

https://github.com/narrowlink/ipstack/commit/670226a9565655c8dcbc927582012e48b2c56580

**Comment by xmh0511** [comment]
I think providing the raw packet produced bu tun for all  the other cases other than tcp and udp has the maximum flexible because we may want to do the following things:
> 1. process the packet by using the familiar crate.
> 2. send the **complete packet** to the remote to be processed.

**Comment by SajjadPourali** [comment]
@xmh0511: could you please check examples/tun.rs in the main branch?

https://github.com/narrowlink/ipstack/blob/main/examples/tun.rs#L113C1-L134

**Comment by SajjadPourali** [comment]
Screenshot 2024-01-20 at 8 06 59 PM

**Comment by xmh0511** [comment]
@SajjadPourali  I have tried the latest version. Currently, in the branch `IpStackStream::UnknownTransport(u)`, `u.payload()` will return the payload part of the IP packet, which is the ICMP message, and `u.send` just need the ICMP message too, right?

**Comment by SajjadPourali** [comment]
> @SajjadPourali I have tried the latest version. Currently, in the branch `IpStackStream::UnknownTransport(u)`, `u.payload()` will return the payload part of the IP packet, which is the ICMP message, and `u.send` just need the ICMP message too, right?

True, there also another one that called UnknownNetwork that is similar to the raw that you mentioned.

**Pull request closed without merging**